### DISCUSSION TITLE
How would one go about achieving server side window borders?

### DISCUSSION THREAD

**Discussion by mWalrus** [Smithay/smithay#1619] [Q&A] [answered]
I want to decorate windows in my compositor with the classic window borders you can find in most X window managers/wayland compositors. Is there a way for me to achieve this easily?

**Comment by Drakulix** [comment, marked as answer]
Depends on what you consider easy. smithay doesn't give you any pre-made UI elements. You need to integrate whichever toolkit you choose to produce something you can plug into smithay's rendering pipeline and inject input. For an example see [smithay-egui](https://github.com/smithay/smithay-egui).

**Comment by mWalrus** [comment]
Alright, thanks for the reply! I'm starting to realize how much larger the scope is when writing a compositor for wayland compared to an X window manager. It seems like I might have to go back to reading docs.
